apis/helpers/namer_test.go

```diff
@@ -7,56 +7,75 @@ import (
 	"k8s.io/apimachinery/pkg/util/validation"
 )
 
+const (
+	shortSuffix = "deploy"
+	// hashLen is 10 to represent "-" + hash + "-" (hash itself is always 8 characters)
+	hashLen = 10
+)
+
 func TestGetName(t *testing.T) {
 	for i := 0; i < 10; i++ {
-		shortName := randSeq(rand.Intn(validation.DNS1123SubdomainMaxLength-1) + 1)
-		longName := randSeq(validation.DNS1123SubdomainMaxLength + rand.Intn(100))
+		// shortBase + "-" + hash will be underneath the maximum limit without any truncation
+		shortBase := randSeq(rand.Intn(validation.DNS1123SubdomainMaxLength-hashLen) + 1)
+
+		// Any usage of longBase/longSuffix will definitely need to be truncated
+		longBase := randSeq(validation.DNS1123SubdomainMaxLength + 100)
+		longSuffix := randSeq(validation.DNS1123SubdomainMaxLength + 100)
```

Comment:
These could also continue to use `rand.Intn(100)` (possibly `+ 1`?) so they're also variable. Being at minimum `validation.DNS1123SubdomainMaxLength + 1` will ensure we get truncation behavior.

Reply:
Works for me, yeah `rand.Intn(x)` generates a range from `[0, x)`, so you're right about the +1 to guarantee that we go over `validation.DNS1123SubdomainMaxLength` haha